common/unsafe/src/main/java/org/apache/spark/unsafe/memory/HeapMemoryAllocator.java

```diff
@@ -46,9 +46,12 @@ private boolean shouldPool(long size) {
 
   @Override
   public MemoryBlock allocate(long size) throws OutOfMemoryError {
-    if (shouldPool(size)) {
+    int numWords = (int) ((size + 7) / 8);
```

Comment:
let's add some check to make sure `(size + 7) / 8` doesn't exceed max int.

Reply:
L51: assert (alignedSize >= size); 
I think `assert (alignedSize >= size) ` can make sure `(size + 7) / 8` doesn't exceed max int.